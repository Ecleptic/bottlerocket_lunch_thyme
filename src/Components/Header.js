import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import logo from '../assets/logo.svg'
import backImg from '../assets/ic_webBack@2x.png'
import mapImg from '../assets/icon_map@2x.png'

const StyledHeader = styled.header`
    align-items: center;
    background: #43e895;
    display: flex;
    height: 80px;
    justify-content: space-around;
    padding-top: 50px;
    padding-bottom: 20px;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 3;
    @media (min-width: 640px) {
        height: 80px;
    }
`

const Logo = styled.img`
    height: 17px;
    @media (min-width: 640px) {
        height: 50px;
    }
`
const Image = styled.img`
    height: 30px;
    padding-bottom: 5px;
    @media (min-width: 640px) {
        height: 50px;
    }
`
const Anchor = styled.a`
    color: #2a2a2a;
    text-decoration: none;
`

export default function Header({ isMobile, details, detailIsRendered }) {
    const mapsLink = () => {
        let locationString = ''
        if (details !== null && detailIsRendered) {
            const lat = details.location.lat
            const lng = details.location.lng
            locationString = `?daddr=${lat},${lng}&amp;ll=`
        }

        navigator.platform.indexOf('iPhone') !== -1 ||
        navigator.platform.indexOf('iPad') !== -1
            ? window.open(`maps://maps.google.com/maps${locationString}`)
            : window.open(`https://maps.google.com/maps${locationString}`)
    }

    return (
        <StyledHeader>
            <Link to={process.env.PUBLIC_URL + '/'}>
                {isMobile &&
                    detailIsRendered && (
                        <Image src={backImg} alt="Back Button" />
                    )}
            </Link>
            <Link to={process.env.PUBLIC_URL + '/'}>
                <Logo src={logo} className="App-logo" alt="logo" />
            </Link>

            <Anchor
                onClick={() => {
                    mapsLink()
                }}
            >
                <Image
                    src={mapImg}
                    alt="Link to Maps"
                />
            </Anchor>
        </StyledHeader>
    )
}

Header.propTypes = {
    detailIsRendered: PropTypes.bool,
    details: PropTypes.object,
    isMobile: PropTypes.bool
}
