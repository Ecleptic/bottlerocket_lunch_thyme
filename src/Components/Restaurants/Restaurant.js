import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import React from 'react'
import styled from 'styled-components'

const Restaurant = ({ restaurant, slug }) => {
    const categoryTitleStyles = `color: #ffffff;
        font-size: 20px;
        margin-bottom: 10px;
        margin-top: 10px;
        padding-left: 12px;
        position: absolute;
        text-shadow: 0 2px 2px rgba(0, 0, 0, 0.5);
        z-index: 2;`

    const Title = styled.h2`
        bottom: 24px;
        font-weight: bold;
        ${categoryTitleStyles};
    `
    const Category = styled.p`
        bottom: 6px;
        ${categoryTitleStyles};
    `

    const StyledLink = styled(Link)`
        display: block;
        height: 180px;
        overflow: hidden;
        position: relative;
        width: 100%;
        @media (min-width: 640px) {
            height: auto;
        }
    `

    const Gradient = styled.div`
        bottom: 0;
        display: flex;
        height: 180px;
        width: 100%;
        z-index: 1;
        background: linear-gradient(
            to bottom,
            rgba(0, 0, 0, 0) 10%,
            rgba(0, 0, 0, 0) 10%,
            rgba(0, 0, 0, 0.65) 80%
        );
        @media (min-width: 640px) {
            height: auto;
        }
    `

    const Image = styled.img`
        height: 180px;
        left: 0;
        position: relative;
        top: 0;
        width: 100%;
        z-index: -1;
        @media (min-width: 640px) {
            height: auto;
        }
    `

    return (
        <StyledLink to={slug}>
            <Gradient>
                <Image
                    src={restaurant.backgroundImageURL}
                    alt={restaurant.name}
                />
                <Title>{restaurant.name}</Title>
                <Category>{restaurant.category}</Category>
            </Gradient>
        </StyledLink>
    )
}

export default Restaurant

Restaurant.propTypes = {
    restaurant: PropTypes.object
}
