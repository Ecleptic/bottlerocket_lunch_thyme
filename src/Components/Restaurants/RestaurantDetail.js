import PropTypes from 'prop-types'
import React, { Component } from 'react'

import { MyContext } from '../../Context'
import MapContainer from '../Maps/MapContainer'
import styled from 'styled-components'

export default class RestaurantDetail extends Component {
    componentWillMount() {
        this.props.detailIsRendered(true)
        this.props.getCurrentRestaurant(this.props.match.params.id)
    }

    componentDidUpdate(nextProps) {
        if (nextProps.match.params.id !== this.props.match.params.id) {
            this.props.getCurrentRestaurant(this.props.match.params.id)
        }
    }

    componentWillUnmount() {
        this.props.detailIsRendered(false)
    }

    render() {
        const ParentDiv = styled.div`
            height: 300;
            width: 100%;
            display: flex;
            flex-direction: column;
            padding-top: 50px;
            @media (min-width: 640px) {
                padding-top: 80px;
            }
        `

        const TitleDiv = styled.div`
            background: #34b379;
            color: #ffffff;
            margin: 0;
            height: 60px;
            display: flex;
            flex-direction: column;
            justify-content: center;
            padding-left: 12px;
            h1 {
                font-size: 16px;
                font-weight: bold;
            }
            h3 {
                font-size: 12px;
            }
        `

        const RestaurantInfo = styled.div`
            padding: 16px;
            p {
                margin-bottom: 26px;
            }
        `

        const Anchor = styled.a`
            color: #2a2a2a;
            text-decoration: none;
        `

        return (
            <MyContext.Consumer>
                {context =>
                    context.state.details ? (
                        <ParentDiv context={context}>
                            <TitleDiv>
                                <h1>{context.state.details.name}</h1>
                                <h3>{context.state.details.category}</h3>
                            </TitleDiv>

                            <div style={{ height: '300px', order: -1 }}>
                                {context.state.details.location && (
                                    <MapContainer
                                        style={{ height: '300px' }}
                                        lat={context.state.details.location.lat}
                                        lng={context.state.details.location.lng}
                                        info={{
                                            name: context.state.details.name,
                                            address:
                                                context.state.details
                                                    .formattedAddress
                                        }}
                                    />
                                )}
                            </div>

                            <RestaurantInfo>
                                {context.state.details.location && (
                                    <p>
                                        {context.state.details.location.address}
                                        <br />
                                        {context.state.details.location.city} ,
                                        {context.state.details.location.state}
                                        {
                                            context.state.details.location
                                                .postalCode
                                        }
                                    </p>
                                )}

                                {context.state.details.contact && (
                                    <a
                                        href={`tel:${
                                            context.state.details.contact
                                                .formattedPhone
                                        }`}
                                    >
                                        <p>
                                            {
                                                context.state.details.contact
                                                    .formattedPhone
                                            }
                                        </p>
                                    </a>
                                )}

                                {context.state.details.contact &&
                                    context.state.details.contact.twitter && (
                                        <Anchor
                                            target="_blank"
                                            href={`https://twitter.com/${
                                                context.state.details.contact
                                                    .twitter
                                            }`}
                                        >
                                            <p>
                                                @{
                                                    context.state.details
                                                        .contact.twitter
                                                }
                                            </p>
                                        </Anchor>
                                    )}
                            </RestaurantInfo>
                        </ParentDiv>
                    ) : null
                }
            </MyContext.Consumer>
        )
    }
}

RestaurantDetail.propTypes = {
    detailIsRendered: PropTypes.func,
    getCurrentRestaurant: PropTypes.func,
    match: PropTypes.array
}
