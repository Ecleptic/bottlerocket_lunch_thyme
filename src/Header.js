import { Link } from 'react-router-dom'
import React from 'react'
import styled from 'styled-components'

import logo from './assets/logo.svg'

const StyledHeader = styled.header`
    align-items: center;
    background: #43e895;
    display: flex;
    height: 50px;
    justify-content: center;
    padding-top: 10px;
    position: fixed;
    top:0;
    width: 100%;
    z-index: 3;
    @media (min-width: 640px) {
        height: 80px;
    }
`

const Image = styled.img`
    height: 17px;

    @media (min-width: 640px) {
        height: 50px;
    }
`

const Header = () => {
    return (
        <StyledHeader>
            <Link to={process.env.PUBLIC_URL + '/'}>
                <Image src={logo} className="App-logo" alt="logo" />
            </Link>
        </StyledHeader>
    )
}

export default Header
