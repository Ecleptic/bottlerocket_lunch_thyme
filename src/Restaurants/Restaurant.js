import { connect } from "react-redux"
import { Link } from "react-router-dom"
import React from "react"
import styled from "styled-components"

import { slugify } from "./actions"

const originHeight = 180
const originWidth = 320
const originFontSize = 12
const originMarginSize = 6

const Restaurant = ({ restaurant, windowWidth }) => {
    const computedWidth =
        windowWidth < 640
            ? `${100}%`
            : `${originWidth * 2}px` || `${originWidth * 1.6}px`

    const computedHeight =
        windowWidth < 640
            ? `${originHeight * 1}px`
            : `${originHeight * 2}px` || `${originHeight * 1.6}px`

    const StyledLink = styled(Link)`
        position: relative;
        display: block;
        overflow: hidden;
        width: ${computedWidth};
        height: ${computedHeight};
    `

    const Gradient = styled.div`
        display: inline-block;
        background: linear-gradient(
            to bottom,
            rgba(0, 0, 0, 0) 10%,
            rgba(0, 0, 0, 0) 10%,
            rgba(0, 0, 0, 0.65) 80%
        );
        z-index: 1;
        height: ${computedHeight};
        width: ${computedWidth};
    `
    const Image = styled.img`
        position: relative;
        z-index: -1;
        top: 0;
        left: 0;
        height: ${computedHeight};
        width: ${computedWidth};
    `
    const Title = styled.p`
        position: absolute;
        bottom: ${originMarginSize * 4}px;
        color: #ffffff;
        text-shadow: 0 2px 2px rgba(0, 0, 0, 0.5);
        z-index: 2;
        font-weight: bold;
        font-size: ${originFontSize * 1.6}px;
        margin: ${originMarginSize * 1.6}px, 0;
        padding-left: 12px;
    `
    const Category = styled.p`
        position: absolute;
        bottom: ${originMarginSize}px;
        color: #ffffff;
        text-shadow: 0 2px 2px rgba(0, 0, 0, 0.5);
        z-index: 2;
        font-size: ${originFontSize * 1.6}px;
        margin: ${originMarginSize * 1.6}px, 0;
        padding-left: 12px;
    `

    const slug = slugify(restaurant.name)
    return (
        <StyledLink to={slug}>
            <Gradient>
                <Image
                    src={restaurant.backgroundImageURL}
                    alt={restaurant.name}
                />
                <Title>{restaurant.name}</Title>
                <Category>{restaurant.category}</Category>
            </Gradient>
        </StyledLink>
    )
}

const mapStateToProps = state => ({
    windowWidth: state.restaurants.windowHeight,
    windowHeight: state.restaurants.windowWidth
})

export default connect(mapStateToProps)(Restaurant)
