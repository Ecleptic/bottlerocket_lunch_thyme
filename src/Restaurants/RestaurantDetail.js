import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import React, { Component } from 'react'

import MapContainer from '../Maps/MapContainer'
import styled from 'styled-components'

class RestaurantDetail extends Component {
    state = {
        details: {}
    }

    getCurrentRestaurant = () => {
        const details = this.props.restaurants.restaurants.find(e => {
            return e.name === this.props.lookupTable[this.props.match.params.id]
        })
        this.setState({ details })
    }

    componentDidUpdate(nextProps) {
        if (nextProps.match.params.id !== this.props.match.params.id) {
            this.getCurrentRestaurant()
        }
    }

    componentWillMount() {
        this.getCurrentRestaurant()
        this.props.detailIsRendered(true)
    }

    componentWillUnmount() {
        this.props.detailIsRendered(false)
    }

    render() {
        const ParentDiv = styled.div`
            height: 300;
            width: 100%;
            display: flex;
            flex-direction: column;
            padding-top: 50px;
            @media (min-width: 640px) {
                height: 80px;
            }
        `

        const TitleDiv = styled.div`
            background: #34b379;
            color: #ffffff;
            margin: 0;
            height: 60px;
            display: flex;
            flex-direction: column;
            justify-content: center;
            padding-left: 12px;
            h1 {
                font-size: 16px;
                font-weight: bold;
            }
            h3 {
                font-size: 12px;
            }
        `
        const RestaurantInfo = styled.div`
            padding: 16px;
            p {
                margin-bottom: 26px;
            }
        `
        const Anchor = styled.a`
            color: #2a2a2a;
            text-decoration: none;
        `

        if (this.state.details) {
            return (
                <ParentDiv>
                    <TitleDiv>
                        <h1>{this.state.details.name}</h1>
                        <h3>{this.state.details.category}</h3>
                    </TitleDiv>

                    <div style={{ height: '300px', order: -1 }}>
                        {this.state.details.location && (
                            <MapContainer
                                style={{ height: '300px' }}
                                lat={this.state.details.location.lat}
                                lng={this.state.details.location.lng}
                                info={{
                                    name: this.state.details.name,
                                    address: this.state.details.formattedAddress
                                }}
                            />
                        )}
                    </div>

                    <RestaurantInfo>
                        {this.state.details.location && (
                            <p>
                                {this.state.details.location.address}
                                <br />
                                {this.state.details.location.city} ,
                                {this.state.details.location.state}
                                {this.state.details.location.postalCode}
                            </p>
                        )}

                        {this.state.details.contact && (
                            <a
                                href={`tel:${
                                    this.state.details.contact.formattedPhone
                                }`}
                            >
                                <p>
                                    {this.state.details.contact.formattedPhone}
                                </p>
                            </a>
                        )}

                        {this.state.details.contact &&
                            this.state.details.contact.twitter && (
                                <Anchor
                                    target="_blank"
                                    href={`https://twitter.com/${
                                        this.state.details.contact.twitter
                                    }`}
                                >
                                    <p>@{this.state.details.contact.twitter}</p>
                                </Anchor>
                            )}
                    </RestaurantInfo>
                </ParentDiv>
            )
        } else {
            return null
        }
    }
}
const mapStateToProps = state => ({
    lookupTable: state.restaurants.lookupTable,
    restaurants: state.restaurants.restaurants
})

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            // getCurrentRestaurant
        },
        dispatch
    )

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RestaurantDetail)
