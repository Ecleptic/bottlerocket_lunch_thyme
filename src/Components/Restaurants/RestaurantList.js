import { bindActionCreators } from "redux"
import { connect } from "react-redux"
import React, { Component } from "react"

import { getRestaurants } from "./actions"
import Restaurant from "./Restaurant"

class RestaurantList extends Component {
    componentWillMount() {
        const { getRestaurants } = this.props
        getRestaurants()
    }

    render() {
        const { restaurants } = this.props.restaurants
        return (
            <div className="restaurantList">
                {restaurants && restaurants.map((restaurant, index) => (
                    <Restaurant key={index} restaurant={restaurant} />
                ))}
            </div>
        )
    }
}

const mapStateToProps = state => ({
    restaurants: state.restaurants.restaurants,
    currentRestaurant: state.restaurants.currentRestaurant

})

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            getRestaurants
        },
        dispatch
    )

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RestaurantList)
