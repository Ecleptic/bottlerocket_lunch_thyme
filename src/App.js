import { applyMiddleware, createStore } from 'redux'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { composeWithDevTools } from 'redux-devtools-extension'
import { Provider } from 'react-redux'
import { save, load } from 'redux-localstorage-simple'
import logger from 'redux-logger'
import React, { Component } from 'react'
import thunk from 'redux-thunk'

import Header from './Components/Header'
import RestaurantDetail from './Components/Restaurants/RestaurantDetail'
import RestaurantList from './Components/Restaurants/RestaurantList'
import rootReducer from './rootReducer'

const middleware = [logger, thunk]

const store = createStore(
    rootReducer,
    load(),
    composeWithDevTools(applyMiddleware(...middleware, save()))
)

class App extends Component {
    state = {
        currentRestaurant: '',
        detailIsRendered: false,
        isMobile: true
    }

    componentWillMount = () => {
        this.updateDimensions()
        window.addEventListener('resize', this.updateDimensions)
    }

    componentWillUnmount() {
        window.removeEventListener('resize', this.updateDimensions)
    }

    detailIsRendered = detailIsRendered => {
        console.log('detailIsRendered', detailIsRendered)
        this.setState({ detailIsRendered })
    }

    updateDimensions = () => {
        if (window.matchMedia('(min-width: 640px)').matches) {
            this.setState({ isMobile: false })
        } else {
            this.setState({ isMobile: true })
        }
    }

    render() {
        return (
            <Provider store={store}>
                <Router>
                    <div className="App">
                        <Header />
                        <div className="appContainer">
                            <RenderList
                                detailIsRendered={this.state.detailIsRendered}
                                isMobile={this.state.isMobile}
                            />
                            {!this.state.detailIsRendered &&
                                !this.state.isMobile && (
                                    <h1 style={{ paddingTop: '80px' }}>
                                        Please Select a Restaurant
                                    </h1>
                                )}
                            <Switch>
                                <Route
                                    path={`${process.env.PUBLIC_URL}/:id`}
                                    render={props => (
                                        <RestaurantDetail
                                            {...props}
                                            detailIsRendered={
                                                this.detailIsRendered
                                            }
                                        />
                                    )}
                                />
                            </Switch>
                        </div>
                    </div>
                </Router>
            </Provider>
        )
    }
}

const RenderList = ({ detailIsRendered, isMobile }) => {
    if (!isMobile) {
        return <RestaurantList />
    } else {
        if (!detailIsRendered) {
            return <RestaurantList />
        } else {
            return null
        }
    }
}

export default App
