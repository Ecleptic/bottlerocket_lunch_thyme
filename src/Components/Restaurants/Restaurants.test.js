import React from 'react'

import App from '../../App'
import Restaurant from './Restaurant'
import RestaurantDetail from './RestaurantDetail'
import RestaurantList from './RestaurantList'
import Header from '../Header'

describe('All View components match snapshots', () => {
    it('matches the snapshot', () => {
        const app = <App />
        expect(app).toMatchSnapshot()
    })

    it('matches the snapshot', () => {
        const list = <RestaurantList />
        expect(list).toMatchSnapshot()
    })

    it('matches the snapshot', () => {
        const RestaurantComponent = <Restaurant />
        expect(RestaurantComponent).toMatchSnapshot()
    })

    it('matches the snapshot', () => {
        const detail = <RestaurantDetail />
        expect(detail).toMatchSnapshot()
    })
    it('matches the snapshot', () => {
        const head = <Header />
        expect(head).toMatchSnapshot()
    })
})
