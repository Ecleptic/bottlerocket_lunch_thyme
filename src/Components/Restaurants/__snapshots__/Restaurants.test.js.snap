// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All View components match snapshots matches the snapshot 1`] = `<App />`;

exports[`All View components match snapshots matches the snapshot 2`] = `<Connect(RestaurantList) />`;

exports[`All View components match snapshots matches the snapshot 3`] = `<Restaurant />`;

exports[`All View components match snapshots matches the snapshot 4`] = `<Connect(RestaurantDetail) />`;

exports[`All View components match snapshots matches the snapshot 5`] = `<Header />`;
